aws_lambda.py

```diff
@@ -77,15 +87,29 @@ def validate_signature(event, context):
             # Some records and empty signature? It will fail below.
             signature = {}
 
-        # 5. Grab the public key
         try:
+            # 5. Verify the signature with the public key
             pubkey = signature['public_key'].encode('utf-8')
             data = b'Content-Signature:\x00' + serialized.encode('utf-8')
             verifier = ecdsa.VerifyingKey.from_pem(pubkey)
-            signature = base64.urlsafe_b64decode(signature['signature'])
-            verified = verifier.verify(signature, data, hashfunc=hashlib.sha384)
+            signature_bytes = base64.urlsafe_b64decode(signature['signature'])
+            verified = verifier.verify(signature_bytes, data, hashfunc=hashlib.sha384)
             assert verified, "Signature verification failed"
 
+            # 6. Verify that the x5u certificate is valid (ie. that signature was well refreshed)
+            x5u = signature['x5u']
+            if x5u not in checked_certificates:
+                resp = requests.get(signature['x5u'])
+                cert_pem = resp.text.encode('utf-8')
+                cert = cryptography.x509.load_pem_x509_certificate(cert_pem, crypto_default_backend())
+                assert cert.signature_algorithm_oid == cryptography.x509.OID_ECDSA_WITH_SHA384, "unexpected signature algorithm"
+                assert cert.not_valid_before < datetime.now(), "certificate not yet valid"
+                assert cert.not_valid_after > datetime.now(), "certificate expired"
+                assert cert.subject.get_attributes_for_oid(NameOID.COMMON_NAME)[0].value.endswith('content-signature.mozilla.org'), "invalid subject name"
```

Comment:
you could also check that the host part of the common name matches the collection name. `onecrl` will have the common name `onecrl.content-signature.mozilla.org`, etc.